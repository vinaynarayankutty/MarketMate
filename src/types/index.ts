export type StockHolding = {
  symbol: string;
  quantity: number;
  ltp: number;
  avgPrice: number;
  close: number;
};
