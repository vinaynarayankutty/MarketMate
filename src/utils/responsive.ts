import {Dimensions} from 'react-native';
const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

export const RPW = (percentage: number) => {
  return (percentage / 100) * screenWidth;
};

export const RPH = (percentage: number) => {
  return (percentage / 100) * screenHeight;
};
