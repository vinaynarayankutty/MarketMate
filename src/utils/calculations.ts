import {StockHolding} from '../types';

export const currentValue = (item: StockHolding) => {
  return item.ltp * item.quantity;
};

export const investmentValue = (item: StockHolding) => {
  return item.avgPrice * item.quantity;
};

export const profitLoss = (item: StockHolding) => {
  return currentValue(item) - investmentValue(item);
};

export const calculateTotalCurrentValue = (
  holdings: StockHolding[],
): number => {
  return holdings.reduce(
    (total, holding) => total + holding.ltp * holding.quantity,
    0,
  );
};

export const calculateTotalInvestmentValue = (
  holdings: StockHolding[],
): number => {
  return holdings.reduce(
    (total, holding) => total + holding.avgPrice * holding.quantity,
    0,
  );
};

export const calculateTotalProfitLoss = (
  totCurrentValue: number,
  totInvestmentValue: number,
): number => totCurrentValue - totInvestmentValue;

export const calculateTodaysProfitLoss = (holdings: StockHolding[]): number => {
  return holdings.reduce(
    (total, holding) =>
      total + (holding.close - holding.ltp) * holding.quantity,
    0,
  );
};
