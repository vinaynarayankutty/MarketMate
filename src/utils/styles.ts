import {StyleSheet} from 'react-native';
import * as Colors from '../constants/colors';
import {RPW} from './responsive';

const Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.WHITE,
  },

  listContainer: {flex: 1},

  contentContainer: {padding: 12},

  summaryContainer: {
    backgroundColor: Colors.WHITE,
    paddingHorizontal: 16,
    borderTopLeftRadius: 8,
    borderTopRightRadius: 8,
    borderWidth: 1,
    borderColor: Colors.PINK,
    shadowColor: Colors.BLACK,
    shadowOffset: {width: 0, height: -3},
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 5,
    justifyContent: 'space-between',
  },

  btnContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    width: RPW(30),
    alignSelf: 'center',
  },

  arrow: {height: RPW(8), width: RPW(8), resizeMode: 'contain'},

  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },

  plContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: 18,
  },

  titleText: {fontSize: 16, fontWeight: 'bold', color: Colors.BLACK},

  valueText: {
    fontSize: 16,
    color: Colors.BLACK,
  },
});

export default Styles;
