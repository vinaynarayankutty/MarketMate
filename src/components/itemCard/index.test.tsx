import React from 'react';
import {render} from '@testing-library/react-native';
import ItemCard from './index';
import {StockHolding} from '../../types';
import {profitLoss} from '../../utils/calculations';

// Mock the profitLoss function
jest.mock('../../utils/calculations', () => ({
  profitLoss: jest.fn().mockReturnValue(100.5),
}));

describe('ItemCard', () => {
  const mockItemData: StockHolding = {
    symbol: 'MAHABANK',
    quantity: 990,
    ltp: 38.05,
    avgPrice: 35,
    close: 40,
  };

  it('renders correctly', () => {
    const {toJSON} = render(
      <ItemCard itemData={mockItemData} index={0} dataLength={5} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('displays the correct stock symbol', () => {
    const {getByTestId} = render(
      <ItemCard itemData={mockItemData} index={0} dataLength={5} />,
    );
    expect(getByTestId('stock-symbol')).toHaveTextContent('MAHABANK');
  });

  it('displays the correct LTP', () => {
    const {getByTestId} = render(
      <ItemCard itemData={mockItemData} index={0} dataLength={5} />,
    );
    expect(getByTestId('ltp-text')).toHaveTextContent('LTP: ₹ 38.05');
  });

  it('displays the correct quantity', () => {
    const {getByTestId} = render(
      <ItemCard itemData={mockItemData} index={0} dataLength={5} />,
    );
    expect(getByTestId('quantity-text')).toHaveTextContent('990');
  });

  it('displays the correct P/L', () => {
    const {getByTestId} = render(
      <ItemCard itemData={mockItemData} index={0} dataLength={5} />,
    );
    expect(getByTestId('pl-text')).toHaveTextContent('P/L: ₹ 100.50');
  });

  it('calls profitLoss function with correct arguments', () => {
    render(<ItemCard itemData={mockItemData} index={0} dataLength={5} />);
    expect(profitLoss).toHaveBeenCalledWith(mockItemData);
  });
});
