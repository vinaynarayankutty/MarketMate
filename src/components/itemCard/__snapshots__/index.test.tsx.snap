// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 0.5,
        "paddingVertical": 10,
      },
      false,
    ]
  }
  testID="item-card-container"
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
      testID="stock-symbol"
    >
      MAHABANK
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
      testID="ltp-text"
    >
      LTP: 
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        ₹ 
        38.05
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
      testID="quantity-text"
    >
      990
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 14,
        }
      }
      testID="pl-text"
    >
      P/L:
       
      <Text
        style={
          {
            "fontWeight": "bold",
          }
        }
      >
        ₹ 
        100.50
      </Text>
    </Text>
  </View>
</View>
`;
